### DISCUSSION TITLE
552th Place (top 39%) in Kaggle Tabular Playground Series Feb 2021

### DISCUSSION THREAD

**Discussion by JamesMcGuigan** [pycaret/pycaret#1096] [Show and tell]
My main objective with this series of competitions was to practice and explore different techniques with minimal effort. With this competition, I discovered a new tool:

PyCaret2

I took one of the example notebooks from the PyCaret2 website, tweaked it to fit the competition dataset, then spent a small amount of time figuring out how to optimize things.

PyCaret2 reminded me of one of my earliest experiments on the Zillow dataset, where I read through the Scikit-learn documentation and created a loop to test and compare all the available algorithms. The difference being that PyCaret2 made the whole thing plug-and-play rather than requiring a week of coding.

My best RMSE score was 0.70011, which is only 0.0063 (= 0.9%) less than the 0.69381 score at the top of the leaderboard. For all practical purposes, that is good enough to use in production, especially given the minimal effort required to achieve this result. PyCaret2 scored 552th place (top 39%).

This highlights a major difference between real-world usage and a competition setting. Where do we draw the line between getting good results quickly and obsessing about minor digits in our accuracy scores?

Overall I was rather impressed by PyCaret2. My biggest complaint was that it was a little slow to run, but that's what a batch job in commit mode is for. My biggest compliment is that it took most of the manual effort out of actually having to do data science. Also if you take the time to read the manual, the API itself almost reads as a tutorial as to best practice for how to deal with tabular data problems.

- https://www.kaggle.com/jamesmcguigan/tps-pycaret2-automl-regression
- https://pycaret.org

**Comment by pycaret** [comment]
@JamesMcGuigan Amazing. Thanks for sharing with the community. Keep supporting!

**Reply by ngupta23** [reply]
Great Job @JamesMcGuigan . The first show and tell under Discussions!
